### PR TITLE
Update Python API casing + add phi-2 and llama samples 

### PR DESCRIPTION
**Pull request by natke**

### CODE REVIEW

File: examples/phi2/model/tokenizer_config.json [hunk 1]
```diff
@@ -0,0 +1,323 @@
+{
```
[the remark this answers is not in the document]
Comment (natke) [reply examples/phi2/model/tokenizer_config.json:1]: I did mean to check them in, in the absence of download instructions. Once we have those we can remove the files

File: examples/phi2/phi2-loop.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+print(prompt, end='', flush=True)
+
+
+while not generator.is_done():
+    generator.compute_logits()
+
+    # search.apply_minLength(1)
+    # search.apply_repetition_penalty(1.0)
+
+    generator.generate_next_token_topp(0.7, 0.6)
```
[the remark this answers is not in the document]
Comment (natke) [reply examples/phi2/phi2-loop.py:33]: Done

File: src/python/python.cpp [hunk 3]
```diff
@@ -266,22 +266,22 @@ PYBIND11_MODULE(onnxruntime_genai, m) {
...
-      .def("Generate", [](Model& model, PySearchParams& search_params) { search_params.Prepare(); return Generate(model, search_params); })
+      .def("generate", [](Model& model, PySearchParams& search_params) { search_params.Prepare(); return Generate(model, search_params); })
 #if USE_TOKENIZER
-      .def("CreateTokenizer", [](Model& model) { return model.CreateTokenizer(); })
+      .def("create_tokenizer", [](Model& model) { return model.CreateTokenizer(); })
 #endif
-      .def_property_readonly("DeviceType", [](const Model& s) { return s.device_type_; });
+      .def_property_readonly("device_type", [](const Model& s) { return s.device_type_; });
 
   pybind11::class_<PyGenerator>(m, "Generator")
```
Comment (RyanUnderhill) [inline src/python/python.cpp:275]: Generator->generator?

File: examples/phi2/phi2-loop.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    Print all primes between 1 and n
+    """'''
+
+input_tokens = tokenizer.encode(prompt)
+
+params=og.search_params(model)
+params.max_length = 256
+params.input_ids = input_tokens
+
+generator=og.Generator(model, params)
```
Comment (RyanUnderhill) [inline examples/phi2/phi2-loop.py:20]: Should this one be uppercase still? (It's uppercase in the python.cpp also but others are lowercase)

File: src/python/python.cpp [hunk 5]
```diff
@@ -77,7 +77,7 @@ void TestFP16(pybind11::array_t<float16> inputs) {
 
 std::string ToString(const GeneratorParams& v) {
   std::ostringstream oss;
-  oss << "SearchParams("
+  oss << "search_params("
```
Comment (RyanUnderhill) [inline src/python/python.cpp:80]: Should this search_params be reverted back to SearchParams too?

**Review by RyanUnderhill — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**